### PR TITLE
[Docs] Adding highlighting section to high level client docs

### PR DESCRIPTION
**Pull request by cbuescher**
This adds a section about how to add highlighting to the SearchSourceBuilder and
how to retrieve highlighted fragments from the SearchResponse.

### CODE REVIEW

File: docs/java-rest/high-level/apis/search.asciidoc [hunk 1]
```diff
@@ -210,6 +230,20 @@ cases need to be cast accordingly:
...
+or many highlighted text fragments:
+
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-request-highlighting-get]
+--------------------------------------------------
+<1> Get the highlighting for the `title` field
+<2> Get one or many fragments containing the highlighted field content
+
```
Comment (javanna) [inline docs/java-rest/high-level/apis/search.asciidoc]: add a link to the section where we show the response part?

**Review by javanna — APPROVED** [review]
LGTM

File: docs/java-rest/high-level/apis/search.asciidoc (hunk 1 above)
Comment (cbuescher) [inline docs/java-rest/high-level/apis/search.asciidoc]: This is the response part, so I assume you mean to add the link to the part above? I did this with the last commit, let me know if you meant something else.

Comment (javanna) [inline docs/java-rest/high-level/apis/search.asciidoc]: thanks that's what I meant.

**Pull request merged**